### DISCUSSION TITLE
Sponsorblock

### DISCUSSION THREAD

**Discussion by lucascompython** [EvieePy/Wavelink#262] [General]
In the README it says that Sponsorblock has built in support, but I don't see any mentions to that in the docs.

**Comment by chillymosh** [comment]
You simply install Sponsorblock on your Lavalink node (or if using a public one it may already have it installed) then you simply send the payload you want for the categories you want to block via https://wavelink.dev/en/latest/wavelink.html#wavelink.Node.send

```
PUT /v4/sessions/{sessionId}/players/{guildId}/sponsorblock/categories

[
  "sponsor",
  "selfpromo"
]
```

**Reply by lucascompython** [reply]
Wow that was fast!
Thank you for the reply!
When should I send that payload? Before every track?

**Reply by chillymosh** [reply]
You only need to send this once on session creation. Most people do it when their bot joins the voice chat.

**Reply by lucascompython** [reply]
Okay thank you again!
I have a final doubt. Is there a way to listen to [these](https://github.com/topi314/Sponsorblock-Plugin#segmentskipped) events?

**Reply by EvieePy** [reply]
I'll add support for unknown events today sometime. But the data won't be presented to you in any pythonic way like the rest of the events, and it will be a generic event name so you will have to determine where it came from and etc yourself. Supporting all generic plugin events as their own event would be too much work for wavelink to keep up with, especially since they don't send any data telling you where the event came from, from my understanding the same event could potentially occur in two different plugins in the current design of Lavalink.

**Reply by lucascompython** [reply]
I am looking into it!
Thank you!
